kots 1.23.0 release notes (#334)
replicatedhq/kots.io: content/release-notes/1.23.0.md

---
date: 2020-11-13T13:30:14-08:00
linktitle: "1.23.0"
title: "1.23.0"
kubernetes: "1.17, 1.18, and 1.19"
---

{{<features>}}

> * Added new template function [KotsVersion](https://kots.io/reference/template-functions/static-context/#kotsversion)

Reviewer: ```suggestion
* Added new template function for [KotsVersion](https://kots.io/reference/template-functions/static-context/#kotsversion)
```

> * Added [collectors](https://troubleshoot.sh/docs/collect/ceph/) and [analyzers](https://troubleshoot.sh/docs/analyze/ceph-status/) for Ceph.

Reviewer: ```suggestion
* Added a [collector](https://troubleshoot.sh/docs/collect/ceph/) and an [analyzer](https://troubleshoot.sh/docs/analyze/ceph-status/) for Ceph.
```

* Added a [collector](https://troubleshoot.sh/docs/collect/collectd/) for Collectd RRD files.
{{</features>}}

{{<changes>}}

> * Added a button for redeploying failed deployments.

Reviewer: @jgruica Is it only for failed deployments? I thought you could redeploy successful deployments too.

Reviewer 2: @markpundsack this is just for failed deployments

Reviewer: Ah, ok. I'd still like to be able to redeploy things we "think" were successful, but weren't somehow. But I'll make a separate CH for that.

> * User will be warned if an incompatible version of [Velero](https://kots.io/vendor/snapshots/overview/#velero-version-compatibility) is installed.

Reviewer: ```suggestion
* Added warning if an incompatible version of [Velero](https://kots.io/vendor/snapshots/overview/#velero-version-compatibility) is installed.
```

* Improved description of what instance and application snapshots do.

> * Better error handling when [app configuration](https://kots.io/vendor/config/config-screen/) is being modified.

Reviewer: Can you elaborate here?

Author: [It's just additional logging.](https://github.com/replicatedhq/kots/commit/1f91a86ba520226e4d78151beba1523d55f077ab)

{{</changes>}}

{{<fixes>}}

> * [kURL](https://kots.io/vendor/packaging/using-tls-certs/) based installs will resume TLS cert monitoring after errors.

Reviewer: Not sure what this means. What's the actual change/fix?

Author: kurl proxy monitors the cert used for TLS termination.  if the cert changes, the new one will be used.  previously an error would cause the monitor to terminate.   [this is the commit.](https://github.com/replicatedhq/kots/commit/24b8327014188c3804b938d5c28bf958518b1a94)

> * Fixed a bug could cause old certificates to be used in [kURL](https://kots.io/vendor/packaging/using-tls-certs/) based installs.

Reviewer: ```suggestion
* Fixed a bug that could cause old certificates to be used in [kURL](https://kots.io/vendor/packaging/using-tls-certs/) based installs.
```

> * Fixed a bug that caused version to not be deployed when [upgrading from an airgap bundle](https://kots.io/kotsadm/updating/updating-kots-apps/#airgapped-installs) with the `deploy` flag specified.

Reviewer: CLI specific, right?

Author: Yes, specifically airgap CLI

> * Fixed a bug that caused version to not be deployed when [upgrading from an airgap bundle](https://kots.io/kotsadm/updating/updating-kots-apps/#airgapped-installs) with the `deploy` flag specified.

Reviewer: ```suggestion
* Fixed a bug that caused updates to not be deployed from the CLI when [upgrading from an airgap bundle](https://kots.io/kotsadm/updating/updating-kots-apps/#airgapped-installs) with the `deploy` flag specified.
```

* Fixed a bug that caused `kotsadm` pod to be evicted during backup.
* Fixed a bug that caused authentication to fail if secret `kotsadm-password` is deleted prior to using the [`kots reset-password`](https://kots.io/kots-cli/reset-password/) command.

> * Fixed the bug that caused airgap [application update using command line](https://kots.io/kotsadm/updating/updating-kots-apps/#airgapped-installs) to print that no updates are available.

Reviewer: ```suggestion
* Fixed a bug that caused airgap [application update using command line](https://kots.io/kotsadm/updating/updating-kots-apps/#airgapped-installs) to print that no updates are available.
```
